src/main/java/frc/robot/classes/TimeOfFlightSensor.java

```diff
@@ -0,0 +1,227 @@
+  private final AtomicBoolean threadRunning = new AtomicBoolean(true);
+
+
+  private void threadMain() {
+    // Using JNI for a non allocating read
+    int port = SerialPortJNI.serialInitializePort((byte)1);
+    SerialPortJNI.serialSetBaudRate(port, 115200);
+    SerialPortJNI.serialSetDataBits(port, (byte)8);
+    SerialPortJNI.serialSetParity(port, (byte)0);
+    SerialPortJNI.serialSetStopBits(port, (byte)10);
+
+    SerialPortJNI.serialSetTimeout(port, 1);
+    SerialPortJNI.serialEnableTermination(port, '\n');
+    
+    //int port = SerialPortJNI.serialInitializePort((byte)3);
+    HAL.report(tResourceType.kResourceType_SerialPort, 2);
+
+    byte[] buffer = new byte[257];
+    SingleCharSequence charSeq = new SingleCharSequence();
+    charSeq.data = buffer;
+    IntRef lastComma = new IntRef();
+
+    int distance0;
+    int distance1;
+
+    while (threadRunning.get()) {
+      int read = SerialPortJNI.serialRead(port, buffer, buffer.length - 1);
+      if (read <= 0) {
+        try {
+          threadLock.lock();
+          this.hasDistance0 = false;
+          this.hasDistance1 = false;
+        } finally {
+          threadLock.unlock();
+        }
+        continue;
+      }
+      if (!threadRunning.get()) {
+        break;
+      }
+
+      // Trim trailing newline if exists
+      if (buffer[read - 1] == '\n') {
+        read--;
+      }
+
+      if (read == 0) {
+        continue;
+      }
+
+      lastComma.value = -1;
+
+      distance0 = parseIntFromIndex(charSeq, read, lastComma);
+      distance1 = parseIntFromIndex(charSeq, read, lastComma);
+      boolean hasDistance0 = distance0 <= 0;
+      boolean hasDistance1 = distance1 <= 0;
+
+      double ts = Timer.getFPGATimestamp();
+
+      try {
+        threadLock.lock();
+        this.lastReadTime = ts;
+        this.hasDistance0 = hasDistance0;
+        this.hasDistance1 = hasDistance1;
+        if (hasDistance0) {
+          this.distance0 = distance0;
+        }
+        if (hasDistance1) {
+          this.distance1 = distance1;
+        }
+      } finally {
+        threadLock.unlock();
+      }
+    }
+
+    SerialPortJNI.serialClose(port);
+  }
+
+  public TimeOfFlightSensor() {
+    readThread = new Thread(this::threadMain);
+    readThread.setName("TimeOfFlightThread");
+    readThread.start();
+  }
+
+  public boolean isSensor0Connected() {
+    try {
+      threadLock.lock();
+      return hasDistance0;
+    } finally {
+      threadLock.unlock();
+    }
+  }
+
+  public boolean isSensor1Connected() {
+    try {
+      threadLock.lock();
+      return hasDistance1;
+    } finally {
+      threadLock.unlock();
+    }
+  }
+
+  public int getDistance0() {
+    try {
+      threadLock.lock();
+      return distance0;
+    } finally {
+      threadLock.unlock();
+    }
+  }
+
+  public void getDistance0(int rawDistance) {
+    try {
+      threadLock.lock();
+      rawDistance = distance0;
+    } finally {
+      threadLock.unlock();
+    }
+  }
```

Comment:
Assigning a parameter to a value has no effect. I don't think this method is useful. Same with getDistance1(int rawDistance)

Reply:
Removed getDistance0(int rawDistance) & getDIstance1(int rawDistance)